external-import/sentinelone-threats/src/s1api.py

```diff
@@ -0,0 +1,124 @@
+import json
+import argparse
+import urllib.parse
+import io
+import zipfile
+from urllib.request import Request, urlopen
+
+
+class SentinelOneApi:
+    """
+    Simple wrapper for SentinelOne API v2.1
+    """
+
+    def __init__(self, api_url, api_token):
+        """
+        api_url: str representing the endpoint to call the SentinelOne API, e.g. https://xxxxxxx.sentinelone.net
+        api_token: str representing the API token
+        """
+        self._api_url = api_url
+        # TODO the api token automatically re-generates after 6 months
```

Comment:
Do you still want to implement the automatic re-generation or do you want to keep it for another time?

Reply:
> Do you still want to implement the automatic re-generation or do you want to keep it for another time?

For now I'm going to leave the comment there, as token re-generation gets complicated, in that it can be re-generated client side, however it must be updated in the data store it is stored in, e.g. AWS Secrets.